lib/new-utils.js

```diff
@@ -8,4 +9,41 @@ function throwNotSupported(name) {
     throw new ReferenceError(`${name} is not supported by our driver`);
 }
 
+const errorTypeMap = {
+    ...customErrors,
+    Error,
+    RangeError,
+    ReferenceError,
+    SyntaxError,
+    TypeError,
+};
+
+/**
+ * A wrapper function to map napi errors to Node.js errors or custom errors.
+ *
```

Comment:
Add an expected use case for this function:
Smh like: "This should be used always when calling napi functions that may throw errors"

As this function will be used in a lot of places, it's better to write a bit more about it. (Maybe also things like motivation for this approach?)